File: pig_latin.rb
# http://programmingpraxis.com/2009/06/02/pig-latin/

VOWELS = ["a","e","i","o","u"]

def initial_consonant(word,vowels)
  suffix = String.new; index = 0
  suffix << word[index]; index += 1 until vowels.include?(word[index].downcase)
  "#{word[index..-1]}-#{suffix}ay"
end

def initial_vowel(word)
  word + "-way"
end

def translate_to_pig_latin(word, vowels=VOWELS)
  vowels.include?(word[0].downcase) ? initial_vowel(word) : initial_consonant(word,vowels)
end

puts translate_to_pig_latin("abracadabra")
puts translate_to_pig_latin("boomshakalaka")

File: pig_latin.swift
// http://programmingpraxis.com/2009/06/02/pig-latin/

func initialVowel(w: String) -> String {
  return w + "-way"
}

func initialConsonant(w: String, vowels: [String]) -> String {
  var suffix = ""
  var index = 0
  while !contains(vowels,w[index]) {
    suffix += w[index]
    index++
  }
  var newString = ""
  while index < count(w) {
    newString += w[index]
    index++
  }
  return "\(newString)-\(suffix)ay" 
}

func pigLatinize(w: String) -> String {
  let vowels = ["a","e","i","o","u"]
  return contains(vowels,w[0]) ? initialVowel(w) : initialConsonant(w,vowels)
}

extension String {
  subscript (i: Int) -> String {
    return String(Array(self)[i])
  }
}

println(pigLatinize("abracadabra"))
println(pigLatinize("boomshakalaka"))
println(pigLatinize("stream"))
